src/main/scala/linting/rule/LintConflictingModuleNames.scala

```diff
@@ -0,0 +1,101 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.linting
+package rule
+
+import firrtl._
+import firrtl.annotations.{IsModule, SingleTargetAnnotation, Target}
+import firrtl.ir._
+import firrtl.options.Dependency
+
+import chisel3.aop.{Aspect, Select}
+import chisel3.RawModule
+
+import scala.collection.mutable
+
+/** Captures the original desired name for a module
+  */
+case class DesiredNameAnnotation(
+  desiredName: String,
+  target: IsModule
+) extends SingleTargetAnnotation[IsModule] {
+  def duplicate(newTarget: IsModule): DesiredNameAnnotation = {
+    this.copy(target = newTarget)
+  }
+}
+
+/** Collects [[DesiredNameAnnotation]]s for [[LintConflictingModuleNames]] to lint
+  */
+case object LintConflictingModuleNamesAspect extends Aspect[RawModule] {
+  def toAnnotation(top: RawModule): AnnotationSeq = {
+    Select.collectDeep(top) {
+      case m: RawModule => DesiredNameAnnotation(m.desiredName, m.toTarget)
+    }.toSeq
+  }
+}
+
+/** This LintRule checks for module name collisions
+  *
+  * Module name collisions occur when different [[Module]]s are annotated with
+  * [[DesiredNameAnnotation]]s that have the same desiredName. Module name
+  * conflicts will cause a [[LintViolation]].
+  */
+final class LintConflictingModuleNames extends LintRule {
+  val recommendedFix: String = "override desiredName based on module parameters ('override def desiredName = \"...\"')"
+
+  val lintName: String = "conflicting-module-names"
+
+  override def optionalPrerequisites: Seq[Dependency[Transform]] =
+    Dependency[RenameDesiredNames] +: super.optionalPrerequisites
```

Comment:
Should this have a dependency on DedupModules? I see that RenameDesiredNames has a prereq on Dedup, maybe just have a comment then that it also relies on Dedup being run?

Reply:
this extends LintRule which has a dependency on Dedup

Reply:
Right, so can you add the comment?